client/v2/tx/tx.go

```diff
@@ -7,85 +7,107 @@ import (
 	"fmt"
 	"os"
 
+	"github.com/cosmos/gogoproto/grpc"
 	"github.com/cosmos/gogoproto/proto"
 	"github.com/spf13/pflag"
 
 	apitxsigning "cosmossdk.io/api/cosmos/tx/signing/v1beta1"
+	clientcontext "cosmossdk.io/client/v2/autocli/context"
+	"cosmossdk.io/client/v2/autocli/print"
 	"cosmossdk.io/client/v2/broadcast"
 	"cosmossdk.io/client/v2/broadcast/comet"
 	"cosmossdk.io/client/v2/internal/account"
 	"cosmossdk.io/core/transaction"
 
-	"github.com/cosmos/cosmos-sdk/client"
 	"github.com/cosmos/cosmos-sdk/client/input"
-	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/codec/types"
 )
 
 // GenerateOrBroadcastTxCLIWithBroadcaster will either generate and print an unsigned transaction
 // or sign it and broadcast it with the specified broadcaster returning an error upon failure.
-func GenerateOrBroadcastTxCLIWithBroadcaster(ctx client.Context, flagSet *pflag.FlagSet, broadcaster broadcast.Broadcaster, msgs ...transaction.Msg) error {
+func GenerateOrBroadcastTxCLIWithBroadcaster(
+	ctx context.Context,
+	conn grpc.ClientConn,
+	broadcaster broadcast.Broadcaster,
+	msgs ...transaction.Msg,
+) error {
```

Comment:
Ditto, I'd return maybe the output instead of requiring a printer. Autocli or the caller can handle the printing.